mp/src/game/server/momentum/mom_generic_bomb.cpp

```diff
@@ -0,0 +1,118 @@
+﻿//========= Copyright Valve Corporation, All rights reserved. ============//
+//
+// Purpose: Exploding bomb
+//
+//=============================================================================//
+
+#include "cbase.h"
+#include "mom_generic_bomb.h"
+#include "baseanimating.h"
+#include "mom_gamerules.h"
+#include "world.h"
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+BEGIN_DATADESC(CMomGenericBomb)
+DEFINE_KEYFIELD(m_flDamage, FIELD_FLOAT, "damage"), DEFINE_KEYFIELD(m_flRadius, FIELD_FLOAT, "radius"),
+    DEFINE_KEYFIELD(m_iHealth, FIELD_INTEGER, "health"),
+    DEFINE_KEYFIELD(m_strExplodeSoundName, FIELD_SOUNDNAME, "sound"),
+
+    DEFINE_OUTPUT(m_Detonate, "OnDetonate"),
+
+    DEFINE_INPUTFUNC(FIELD_VOID, "Detonate", InputDetonate),
+    END_DATADESC()
+
+        PRECACHE_REGISTER(momentum_generic_bomb);
```

Comment:
Spacing. Should be:

```
BEGIN_DATADESC(...)
    DEFINE_
     ...
END_DATADESC();

PRECACHE_REGISTER(...);
```
I typically put a semicolon so auto formatter doesn't mess with the next line.